### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by ewildgoose**
Improve error handling

- Silence some warnings unless debugging
- Improve error when persistence_path missing
- Improve error when no data to plot

### CODE REVIEW

File: lib/mobius/history.ex [hunk 1]
```diff
@@ -2,6 +2,7 @@ defmodule Mobius.History do
   @moduledoc false
 
   @serialization_version 1
+  @debug Application.compile_env(:mobius, :debug, false)
```
Comment (mattludwigs) [inline lib/mobius/history.ex:5]: I think we can remove this and the `if @deubg` condition.

Ideality this done through configuring your application config file like so

```elixir
# config/config.exs

config :logger, level: :debug
```

If you set the level to say `:info` in your conf then at compile time the calls to logger under the `:info` level will be compiled away. So, it is safe to call the `Logger.deubg` (line 77) and then allow the calling application to configure at what level we want to log.

Comment (ewildgoose) [reply lib/mobius/history.ex:5]: My issue is that I develop with :debug log level in general. However the mobius library was spitting out too much stuff by default. 

I don't see that logging this line is terribly useful unless something is going wrong and someone was debugging the library itself. We kind of need a "trace" log level in Elixir. Something you turn on to spew super low level trace info

In my opinion, for now I'd rather lose this debug completely, or have it commented out such that a developer would turn it only only if needed?

What do you think? Leave as is or completely delete the logging?

Comment (mattludwigs) [reply lib/mobius/history.ex:5]: I think we can remove it, at least in this PR.

Curious as to what you're seeing in the logs. I am good with cleaning up Mobius logging (in a different PR) so that it does not spit out too much stuff and make application-level logging hard to parse.

**Comment by ewildgoose** [comment]
Do you mean to get rid of both Logger calls, or just the one you mention above?

Broadly I think libraries should be thoughtful about logging anything, its more of a job for the higher application level? In this specific case I feel it's "trace" logging, ie very low level data that would only be useful to someone debugging the Mobius library itself?

My logic was that this way we have a method to enable library level tracing if needed to debug something, but it's not enabled by default.

Given that it doesn't break anything and only hides some very low level logging, how about committing it as is and send me a definition of whatever we should remove and I'll create a PR for that? Suggestion is just to try and move forwards? Say no!

**Comment by mattludwigs** [comment]
Can we keep the code as is in the history module?

I was trying to remember why we have that particular log in there. It is supposed to be there to pester us about a feature we need to support. Namely, if we receive a  snapshot of the metrics whose timestamp is outside the known time range we just drop it on the floor. The log is there to be explicit about dropping metrics that were configured and thus expected to be tracked by the consuming software. Then I first work on Mobius there were some ideas on how to solve this so that log is supposed to be temporary anyway. 

The context that I use Mobius is with Nerves and the Nerves context logging is essential to understanding how our system is running and debugging things in the past - even in production. So configuring the compile-time log level via module attribute and mix config will make production debugging a bit of a challenge if we want to switch it on during a debug session.

I am curious if you see this log often? From my experience, this only happens at boot time on Nerves devices before the clock is set, but  I am interested in hearing more cases where this log is printed to get a better understanding.

I think we get rid of the log in the registry module as if the event was not configured to be tracked then dropping that incoming event would be working as expected.

In regards to the lower level tracing, I'd probably suggest using another library like [recon](https://hex.pm/packages/recon). 

Sorry to hold up this PR on the logging and being a bit silly about it. I would merge the error handling improvements as is, I am is just a little held up on the logging changes.

**Comment by ewildgoose** [comment]
OK, done. history.ex is unchanged. Have removed the logging completely from registry.ex

We can revisit noisy logging later

**Review by mattludwigs — APPROVED** [review]

**Comment by mattludwigs** [comment]
@ewildgoose thank you!

I am thinking long-term removing the logging and adding some higher-level tracing functions so that I can debug and trace without relying on logging. Plus, if Mobius can provide a high-level tracing API then developers who are interested in Mobius debug do not have to dig into the depths of OTP tracing.
